DGN/Controllers/CategoriesController.cs

```diff
@@ -7,6 +7,7 @@
 using Microsoft.EntityFrameworkCore;
 using DGN.Data;
 using DGN.Models;
+using Microsoft.AspNetCore.Authorization;
 
 namespace DGN.Controllers
 {
```

Comment:
You can just put:
[Authorize(Roles = "Admin")]

above the class and delete the rest

Reply:
no, because there is a function getAll that even unregistered need to have access to.

> GetAll (getting all categories in json) is for everyone because we want to let everyone see getAll articles page that has search with category so it uses it.